tools/windows-node-installer/pkg/cloudprovider/azure/azurevm.go

```diff
@@ -0,0 +1,593 @@
+							Protocol:                 network.SecurityRuleProtocolTCP,
+							SourceAddressPrefix:      to.StringPtr("*"),
+							SourcePortRange:          to.StringPtr("1-65535"),
+							DestinationAddressPrefix: to.StringPtr("*"),
+							DestinationPortRange:     to.StringPtr("3389"),
+							Access:                   network.SecurityRuleAccessAllow,
+							Direction:                network.SecurityRuleDirectionInbound,
+							Priority:                 to.Int32Ptr(100),
+						},
+					},
+					{
+						Name: to.StringPtr("vnet_traffic"),
+						SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
+							Protocol:                 network.SecurityRuleProtocolTCP,
+							SourceAddressPrefix:      to.StringPtr("10.0.0.0/16"),
+							SourcePortRange:          to.StringPtr("*"),
+							DestinationAddressPrefix: to.StringPtr("0.0.0.0/0"),
+							DestinationPortRanges:    &[]string{"1-65535"},
+							Access:                   network.SecurityRuleAccessAllow,
+							Direction:                network.SecurityRuleDirectionInbound,
+							Priority:                 to.Int32Ptr(200),
+						},
+					},
+				},
+			},
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("cannot create security group: %v", err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, az.nsgClient.Client)
+	if err != nil {
+		return fmt.Errorf("cannot create security group: %v", err)
+	}
+
+	return nil
+}
+
+// generateResourceName generates the names for the individual resource components of an instance
+func generateResourceName(resource string) (name string) {
+	name = strings.Join([]string{infraID, "windows", "worker", resource}, "_")
+	return name
+}
+
+// getIPName checks if user has given any IP information if not it will go ahead and
+// create an IP for you. Finally it returns the name of the IP.
+func getIPName(az *AzureProvider) (ipName string, err error) {
+	if az.AzureUserInput.IpName == "" {
+		ipName = generateResourceName("ip")
+		_, err := createPublicIP(context.Background(), az.ipClient, ipName)
+		if err != nil {
+			return "", err
+		}
+		az.AzureUserInput.IpName = ipName
+	} else {
+		ipName = az.AzureUserInput.IpName
+	}
+	return ipName, err
+}
+
+// getvnetProfile gets the vnet Profile of the existing openshift cluster.
+func getvnetProfile(vnetClient network.VirtualNetworksClient) (vnetProfile network.VirtualNetwork) {
+	vnetList, err := vnetClient.List(context.Background(), resourceGroupName)
+	if err != nil {
+		fmt.Errorf("cannot get the vnetProfile info: %v", err)
+		return
+	}
+	vnetListValues := vnetList.Values()
+	vnetProfile = vnetListValues[0]
+	return vnetProfile
+}
+
+// getnsgName gets the network security group name.
+func getnsgName(name *string, az *AzureProvider) (nsgName string, err error) {
+	if name == nil {
+		nsgName = generateResourceName("nsg")
+		err := createSecurityGroup(context.Background(), az, nsgName)
+		if err != nil {
+			return "", err
```

Comment:
like you did here, except err is replaced by fmt.Error("",err)
See AWS implementation for reference.